music_assistant/controllers/player_queues.py

```diff
@@ -1663,6 +1665,22 @@ def _update_queue_from_player(
         queue.available = player.available
         queue.items = len(self._queue_items[queue_id])
 
+        self.logger.debug(
+            "Player %s state update: player state=%s, queue state=%s",
+            player.display_name,
+            player.state,
+            queue.state,
+        )
+
+        if player.state == PlayerState.PAUSED and queue.state != PlayerState.PAUSED:
+            self.logger.info(
+                "Setting queue state to PAUSED to match player state for %s", player.display_name
+            )
+            queue.state = PlayerState.PAUSED
+            self.signal_update(queue_id)
+            return
+
+        # Normal queue state update from player state
```

Comment:
Please do not create mixed PR's.
So please create (or propose) a separate PR for adjusting core functionality and a dedicated PR for adjusting a provider.

Reply:
Also, I don't think the changes in the queue controller are needed this way. I'll have a look because this smells like a one-line fix in the existing update handler.